src/Sequence.cc

```diff
@@ -86,16 +87,20 @@ void Sequence::assign(Sequence::ConstIterator iter, Step&& step)
     enforce_invariants();
 }
 
-void Sequence::check_label(gul14::string_view label)
+std::string Sequence::check_label(gul14::string_view& label)
```

Comment:
Why the reference?

Reply:
Because I wanted to avoid copying.

Reply:
Copying ... a pointer? ;-D
And then the string is copied anyhow by `gul14::trim()`.

I believe all this string optimizations are premature, esp for functions that are never hot like setting the label.

A non-const reference is for out or in/out parameters. I guess `label` is neither.
A const reference could be used to save copying a complex object, which `string_view` isn't.
_Core Guideline [F.16](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rf-in)_

I would have taken a `std::string` directly (letting the copying happen in the calling site), using inplace `trim()` and move the string object out in the end ;-)
We had a similar discussion some time ago (with Olaf?) and there was a nice table that Lars digged out. Can not find it, though.

Reply:
Well, all of these are micro-optimizations, but as a general rule I would:
* Pass `string_view` by value. It consists of two pointers, copying it is usually faster than handling it by reference.

In this specific case, I'd keep `string_view` (without the reference) as the argument type simply because also `trim()` uses it.

Reply:
Fixed: use value parameter instead of pointer.